libass/ass_font.c

```diff
@@ -548,6 +548,29 @@ int ass_face_get_weight(FT_Face face)
     }
 }
 
+FT_Long ass_fsSelection_to_style_flags(uint16_t fsSelection)
+{
+    FT_Long ret = 0;
+
+    if (fsSelection & 1)
+        ret |= FT_STYLE_FLAG_ITALIC;
+    if (fsSelection & (1 << 5))
+        ret |= FT_STYLE_FLAG_BOLD;
+
+    return ret;
+}
+
+FT_Long ass_face_get_style_flags(FT_Face face)
+{
+    // If we have an OS/2 table, compute this ourselves, since FreeType
+    // will mix in some flags that GDI ignores.
+    TT_OS2 *os2 = FT_Get_Sfnt_Table(face, FT_SFNT_OS2);
+    if (os2)
+        return ass_fsSelection_to_style_flags(os2->fsSelection);
+
```

Comment:
GDI also has another edge case.
If there is a [head](https://learn.microsoft.com/en-us/typography/opentype/spec/head) table, it will use the bold/italic of the mac_style.
If will help for mac font like in #679